core/src/main/java/org/elasticsearch/index/mapper/geo/GeoPointFieldMapper.java

```diff
@@ -585,13 +654,17 @@ public GeoPoint decode(long latBits, long lonBits, GeoPoint out) {
 
     private final StringFieldMapper geohashMapper;
 
+    private final boolean bwc;
+
     public GeoPointFieldMapper(String simpleName, MappedFieldType fieldType, MappedFieldType defaultFieldType, Settings indexSettings,
-            ContentPath.Type pathType, DoubleFieldMapper latMapper, DoubleFieldMapper lonMapper, StringFieldMapper geohashMapper,MultiFields multiFields) {
+            ContentPath.Type pathType, DoubleFieldMapper latMapper, DoubleFieldMapper lonMapper, StringFieldMapper geohashMapper,
+            MultiFields multiFields, boolean bwc) {
```

Comment:
I don't like this bwc parameter. We already have the settings here, so you can grab Index.versionCreated. Then name the member accordingly (see FieldMapper base class for example of the same).